actions/get-charm-paths/src/get_charm_paths.py

```diff
@@ -0,0 +1,62 @@
+#!/bin/env python
+import argparse
+import os
+from pathlib import Path
+
+
+OUTPUT_VARIABLE_NAME = "charm_paths"
+
+def find_charms_in_dir(base_dir: str = "./", charms_subdir="charms"):
+    """
+    Finds paths to charm directories in base_dir.
+
+    Directories are identified as charm directories by the presence of a "metadata.yaml" or "metadata.yml" file.
+
+    Returns:
+    * If any subdirectories of base_dir/charms_subdir/ are charm directories, then the list of these directories is returned.
+    * If not, then if the base directory itself is a charm directory, that is returned.
+    * Otherwise, returns []
+
+	Directories returned are alphabetically sorted.
+    """
+    base_dir = Path(base_dir)
+    charms_dir = base_dir / charms_subdir
+    
+    # Look for both .yaml and .yml files in subdirectories using a single glob pattern
+    metadata_files = list(charms_dir.glob("*/metadata.y*ml"))
```

Comment:
I think this regex matches much more than "yaml" and "yml" ... e.g yeml or yiml or yoml. But its minor think I still think we can merge.